src/interpreter.cr

```diff
@@ -0,0 +1,95 @@
+
+struct Interpreter
```

Comment:
Why `struct` instead of `class` here? Will it be a significant performance boost?

Reply:
It definitely started out with a search for optimal best practices; but, by the time I'd read [this](https://crystal-lang.org/reference/1.9/guides/performance.html#use-structs-when-possible), I honestly just felt it didn't need the bells & whistles that a `class` garners. Plus, if there comes a need for callers to modify it in the future, it's quick swap of the keyword.